config/config.go

```diff
@@ -8,14 +8,19 @@ import (
 	"github.com/spf13/viper"
 )
 
-func NewConfig() Configuration {
+func NewConfig(filename string) Configuration {
 	v := viper.New()
 	setConfigDefaults(v)
-	setConfigFile(v)
+
+	if filename == "" {
+		log.Infof("No configuration file was specified, Prebid Cache will initialize with default values")
+	}
+
```

Comment:
Should we not extend this if to contain up to line 24? If filename is intentionally blank, we do not expect `setConfigFile()` to do anything and we can continue to have a fatal error if we tried to read from a config file and that read failed. Probably need to move `setEnvVars(v)` out of this if block though. I fear that removing the fatal error in the case that the host wanted to read from a file would be bad. If an updated file broke the processing, or if an unrelated mishap deleted the config file, we would want immediate attention to the problem. Most people do not read the log output checking for a message like this when deploying prebid cache.

Reply:
Makes perfect sense. Modified to error-out when specified configuration file could not be read or didn't have the right extension.